fuse/plugins/micro_physics/input.py

```diff
@@ -389,29 +388,41 @@ def _load_root_file(self):
             if self.cut_by_eventid:
                 cutby_string = "g4 eventid"
 
-            if self.kwargs["entry_start"] is not None:
-                log.debug(f'Starting to read from {cutby_string} {self.kwargs["entry_start"]}')
-            if self.kwargs["entry_stop"] is not None:
-                log.debug(f'Ending read in at {cutby_string} {self.kwargs["entry_stop"]}')
+            if self.entry_start is not None:
+                log.debug(f"Starting to read from {cutby_string} {self.entry_start}")
+            if self.entry_stop is not None:
+                log.debug(f"Ending read in at {cutby_string} {self.entry_stop}")
 
-        # If user specified entry start/stop we have to update number of
-        # events for source rate computation:
-        if self.kwargs["entry_start"] is not None:
-            start = self.kwargs["entry_start"]
+        # If we cut by eventid we have to read all of them first to find the start and stop index
+        if self.cut_by_eventid:
+            all_eventids = ttree.arrays("eventid")
+
+            if self.entry_start is not None:
+                start_index = np.argmax(all_eventids["eventid"] >= self.entry_start)
```

Comment:
I think you mean `start_index = np.argmax(all_eventids[all_eventids["eventid"] >= self.entry_start])`? Here you are performing `np.argmax` on a boolean array.